tf/upgrade.py

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python3
+import argparse
+import os
+import yaml
+import sys
+import tensorflow as tf
+from tfprocess import TFProcess
+
+START_FROM = 0
+
+def main(cmd):
+    cfg = yaml.safe_load(cmd.cfg.read())
+    print(yaml.dump(cfg, default_flow_style=False))
+
+    root_dir = os.path.join(cfg['training']['path'], cfg['name'])
+    if not os.path.exists(root_dir):
+        os.makedirs(root_dir)
+
+    x = [
+        tf.placeholder(tf.float32, [None, 112, 8*8]),
+        tf.placeholder(tf.float32, [None, 1858]),
+        tf.placeholder(tf.float32, [None, 1])
+    ]
+
+    tfprocess = TFProcess(cfg)
+    tfprocess.init_net(x)
+
+    if os.path.exists(os.path.join(root_dir, 'checkpoint')):
+        cp = tf.train.latest_checkpoint(root_dir)
+        reader = tf.train.NewCheckpointReader(cp)
+        saved_shapes = reader.get_variable_to_shape_map()
+        new_names = sorted(
+            [(var.name, var.name.split(':')[0]) for var in tf.global_variables()
```

Comment:
First element of this tuple doesn't seem to be used, other than being accessed but ignored in the loop below.

Reply:
yeap, I was just being lazy, cleaned it up.